lock/src/main/java/com/auth0/android/lock/Configuration.java

```diff
@@ -172,27 +183,69 @@ private Strategy filterStrategy(Strategy strategy, Set<String> connections) {
         return filtered;
     }
 
+    private List<Strategy> filterPasswordlessStrategies(List<Strategy> strategies, Set<String> connections) {
+        if (strategies == null || connections.isEmpty()) {
+            return strategies;
+        }
+        List<Strategy> filtered = new ArrayList<>(strategies.size());
+        for (Strategy strategy : strategies) {
+            if (connections.contains(strategy.getName())) {
+                filtered.add(strategy);
+            }
+        }
+        return filtered;
+    }
+
     private void parseLocalOptions(Options options) {
         usernameStyle = options.usernameStyle();
         loginAfterSignUp = options.loginAfterSignUp();
 
-        if (getDefaultDatabaseConnection() == null) {
-            return;
-        }
+        if (getDefaultDatabaseConnection() != null) {
+            //let user disable signUp only if connection have enabled it.
+            signUpEnabled = getDefaultDatabaseConnection().booleanForKey(SHOW_SIGNUP_KEY);
+            if (signUpEnabled && !options.isSignUpEnabled()) {
+                signUpEnabled = false;
+            }
 
-        //let user disable signUp only if connection have enabled it.
-        signUpEnabled = getDefaultDatabaseConnection().booleanForKey(SHOW_SIGNUP_KEY);
-        if (signUpEnabled && !options.isSignUpEnabled()) {
-            signUpEnabled = false;
-        }
+            //let user disable signUp only if connection have enabled it.
+            changePasswordEnabled = getDefaultDatabaseConnection().booleanForKey(SHOW_FORGOT_KEY);
+            if (changePasswordEnabled && !options.isChangePasswordEnabled()) {
+                changePasswordEnabled = false;
+            }
 
-        //let user disable signUp only if connection have enabled it.
-        changePasswordEnabled = getDefaultDatabaseConnection().booleanForKey(SHOW_FORGOT_KEY);
-        if (changePasswordEnabled && !options.isChangePasswordEnabled()) {
-            changePasswordEnabled = false;
+            usernameRequired = getDefaultDatabaseConnection().booleanForKey(REQUIRES_USERNAME_KEY);
         }
 
-        usernameRequired = getDefaultDatabaseConnection().booleanForKey(REQUIRES_USERNAME_KEY);
+        PasswordlessType passwordlessType = options.passwordlessType();
+        if (getPasswordlessStrategies().isEmpty()) {
+            passwordlessMode = null;
+            passwordlessType = null;
+        }
+        if (passwordlessType != null) {
```

Comment:
We should start thinking in connection names like we do for the other strategies. We should do the same auth0 strategy does, pick the first connection